include/kernel.h

```diff
@@ -587,6 +616,22 @@ extern FUNC_NORETURN void k_thread_user_mode_enter(k_thread_entry_t entry,
 						   void *p3);
 
 /**
+ * @brief Grant a thread access to a NULL-terminated  set of kernel objects
+ *
+ * This is a convenience function. For the provided thread, grant access to
+ * the remaining arguments, which should be pointers to kernel objects.
+ * The final argument must be a NULL.
+ *
+ * The thread object must be initialized (i.e. running). The objects don't
+ * need to be.
```

Comment:
Might benefit from a comment stating that this shouldn't be a syscall, and explaining the reason.

Reply:
i think that's beyond the scope of function doc strings to be honest.
otherwise, I'd have to document this for every API in kernel.h that is not a system call.

Reply:
Also, once we have a runtime library established, all APIs which are callable from either mode which are not system calls are getting moved to the runtime library, including this API.